code/game/objects/structures/flags.dm

```diff
@@ -0,0 +1,730 @@
+/obj/structure/sign/flag
+	name = "blank banner"
+	desc = "A blank blue flag"
+	icon_state = "flag"
+	/// Is the flag ripped/torn or not. Used for icon state and checks for allowing retrieval of the flag item.
+	var/torn = FALSE
+	/// Holder for the flag item that is dropped when the flag is pulled down without tearing it.
+	var/obj/item/flag/item_holder
+
+
+/obj/item/flag
+	name = "boxed flag"
+	desc = "A flag neatly folded into a wooden container."
+	icon = 'icons/obj/decals.dmi'
+	icon_state = "flag_boxed"
+	/// Icon state for the deployed flag.
+	var/flag_path
+	/// Whether or not the flag is a large (2 tiles wide) flag.
+	var/flag_size = 0
+
+/obj/structure/sign/flag/left
+	icon_state = "flag_l"
+
+/obj/structure/sign/flag/right
+	icon_state = "flag_r"
+
+
+/obj/item/flag/afterattack(var/atom/A, var/mob/user, var/adjacent, var/clickparams)
+	if (!adjacent)
+		return
+
+	if((!iswall(A) && !istype(A, /obj/structure/window)) || !isturf(user.loc))
+		to_chat(user, SPAN_WARNING("You can't place this here!"))
+		return
+
+	var/placement_dir = get_dir(user, A)
+	if (!(placement_dir in cardinal))
+		to_chat(user, SPAN_WARNING("You must stand directly in front of the location you wish to place that on."))
+		return
+
+	var/obj/structure/sign/flag/P = new(user.loc)
+
+	switch(placement_dir)
+		if(NORTH)
+			P.pixel_y = 32
+		if(SOUTH)
+			P.pixel_y = -32
+		if(EAST)
+			P.pixel_x = 32
+		if(WEST)
+			P.pixel_x = -32
+
+	P.dir = placement_dir
+	if(flag_size)
+		P.icon_state = "[flag_path]_l"
+		var/obj/structure/sign/flag/P2 = new(user.loc)
+		P2.icon_state = "[flag_path]_r"
+		P2.dir = P.dir
+		switch(P2.dir)
+			if(NORTH)
+				P2.pixel_y = P.pixel_y
+				P2.pixel_x = 32
+			if(SOUTH)
+				P2.pixel_y = P.pixel_y
+				P2.pixel_x = 32
+			if(EAST)
+				P2.pixel_x = P.pixel_x
+				P2.pixel_y = -32
+			if(WEST)
+				P2.pixel_x = P.pixel_x
+				P2.pixel_y = 32
+		P2.name = name
+		P2.desc = desc
+		P2.desc_info = desc_info
+		P2.desc_fluff = desc_fluff
+	else
+		P.icon_state = "[flag_path]"
+	P.name = name
+	P.desc = desc
+	P.desc_info = desc_info
+	P.desc_fluff = desc_fluff
+	transfer_fingerprints_to(P)
+	P.item_holder = src
+	forceMove(P)
+
+
+/obj/structure/sign/flag/attack_hand(mob/user as mob)
+	if (user.a_intent == I_HURT && !torn && alert("Do you want to rip \the [src] from its place?","You think...","Yes","No") == "Yes")
+		if(!do_after(user, 2 SECONDS, act_target = src))
+			return 0
+
+		user.visible_message(
+			SPAN_WARNING("\The [user] rips \the [src] in a single, decisive motion!" ),
+			SPAN_WARNING("You rip \the [src] in a single, decisive motion!")
+		)
+		playsound(loc, 'sound/items/poster_ripped.ogg', 100, 1)
+		icon_state = "poster_ripped"
+		name = "ripped flag"
+		desc = "You can't make out anything from the flag's original print. It's ruined."
+		torn = TRUE
+		QDEL_NULL(item_holder)
+		update_icon()
```

Comment:
Self reminder to either remove this or commit to overriding `update_icon()` to check for the state of `torn` instead of updating `icon_state` directly